app/controllers/reactions_controller.rb

```diff
@@ -55,55 +55,13 @@ def index
   def create
     remove_count_cache_key
 
-    if params[:reactable_type] == "Article" && params[:category].in?(Reaction::PRIVILEGED_CATEGORIES)
-      clear_moderator_reactions(
-        params[:reactable_id],
-        params[:reactable_type],
-        current_user,
-        params[:category],
-      )
-    end
-
-    category = params[:category] || "like"
-
-    reaction = Reaction.where(
-      user_id: current_user.id,
-      reactable_id: params[:reactable_id],
-      reactable_type: params[:reactable_type],
-      category: category,
-    ).first
+    result = ReactionToggle.toggle(params, current_user: current_user)
```

Comment:
**non-blocking nitpick (feel free to ignore):** Since most of our service objects all use the `.call` method, would it make sense to switch it up to follow that same pattern?

The service class already has _toggle_ in its name so it would likely still be understood that it would toggle the reactions, but I don't feel strongly about this, specially not to delay this IMO

Reply:
Fwiw, when pairing on this, we wanted to clear a path so that we could get started on the `create` API endpoint, our idea was that we could return later for further refactoring on the `ReactionToggle` class.